File: Data Structures & Algorithms/valid-sudoku/submission-1.cpp
#include <vector>
#include <unordered_set>
using namespace std;

class Solution {
public:
    bool isValidSudoku(vector<vector<char>>& board) {
        vector<unordered_set<char>> rows(9), cols(9), boxes(9);

        for (int r = 0; r < 9; r++) {
            for (int c = 0; c < 9; c++) {
                char v = board[r][c];
                    
                if (v == '.') continue;

                int box = (r / 3) * 3 + (c / 3);

                if (rows[r].count(v) || 
                    cols[c].count(v) || 
                    boxes[box].count(v)) {
                    return false;
                }

                rows[r].insert(v);
                cols[c].insert(v);
                boxes[box].insert(v);
            }
        }
        return true;
    }
};

File: Data Structures & Algorithms/is-anagram/submission-1.js
class Solution {
    /**
     * @param {string} s
     * @param {string} t
     * @return {boolean}
     */
    isAnagram(s, t) {
        if (s.length !== t.length) {
            return false;
        }
        const count = {};
        
        for (let ch of s) {
            count[ch] = (count[ch] || 0) + 1;
        }

        for (let ch of t) {
            if (!count[ch]) {
                return false;
            }
            count[ch]--;
        }
        return true;
    }
}
